src/hooks/useWhiteList.tsx

```diff
@@ -1,29 +1,59 @@
+import { useMemo } from 'react'
 import { JsonRpcProvider, Contract, Network } from 'ethers'
 
 import { networks } from '@/constants/tokenAddresses'
 import KassandraWhitelistAbi from '@/constants/abi/KassandraWhitelist.json'
 
-const useWhiteList = (networkId: number) => {
+function whiteListContract(contract: Contract) {
+  const tokensWhitelist = async (): Promise<string[]> => {
+    const value = await contract.getTokens(0, 100)
+
+    return value
+  }
+
+  const countTokens = async (): Promise<bigint> => {
+    const value = await contract.countTokens()
+
+    return value
+  }
+
+  return {
+    tokensWhitelist,
+    countTokens
+  }
+}
+
+export function whiteList(networkId: number) {
   const networkInfo = networks[networkId]
   const network = new Network(networkInfo.chainName, networkInfo.chainId)
   const readProvider = new JsonRpcProvider(networkInfo.rpc, network, {
     staticNetwork: network
   })
-  const read = new Contract(
-    networks[networkId].whiteList,
+
+  const contract = new Contract(
+    networkInfo.whiteList,
     KassandraWhitelistAbi,
     readProvider
   )
 
-  const tokensWhitelist = async (): Promise<string[]> => {
-    const value = await read.getTokens(0, 100)
+  return whiteListContract(contract)
+}
 
-    return value
-  }
+const useWhiteList = (networkId: number) => {
+  return useMemo(() => {
+    const networkInfo = networks[networkId]
+    const network = new Network(networkInfo.chainName, networkInfo.chainId)
+    const readProvider = new JsonRpcProvider(networkInfo.rpc, network, {
+      staticNetwork: network
+    })
+    const contract = new Contract(
+      networkInfo.whiteList,
+      KassandraWhitelistAbi,
+      readProvider
+    )
 
-  return {
-    tokensWhitelist
-  }
+    return whiteListContract(contract)
+  }, [])
```

Comment:
`networkId` is a dependency

```suggestion
  }, [networkId])
```